src/main/java/com/github/lunatrius/schematica/world/schematic/SchematicFormat.java

```diff
@@ -71,20 +105,63 @@ public static boolean writeToFile(final File file, final ISchematic schematic) {
         return false;
     }
 
-    public static boolean writeToFile(final File directory, final String filename, final ISchematic schematic) {
-        return writeToFile(new File(directory, filename), schematic);
+    /**
+     * Writes the given schematic.
+     *
+     * @param directory The directory to write in
+     * @param filename The filename (including the extension) to write to
+     * @param format The format to use, or null for {@linkplain #FORMAT_DEFAULT the default}
+     * @param schematic The schematic to write
+     * @return True if successful
+     */
+    public static boolean writeToFile(final File directory, final String filename, @Nullable final String format, final ISchematic schematic) {
+        return writeToFile(new File(directory, filename), format, schematic);
     }
 
-    public static void writeToFileAndNotify(final File file, final ISchematic schematic, final EntityPlayer player) {
-        final boolean success = writeToFile(file, schematic);
+    /**
+     * Writes the given schematic, notifying the player when finished.
+     *
+     * @param file The file to write to
+     * @param format The format to use, or null for {@linkplain #FORMAT_DEFAULT the default}
+     * @param schematic The schematic to write
+     * @param player The player to notify
+     */
+    public static void writeToFileAndNotify(final File file, @Nullable final String format, final ISchematic schematic, final EntityPlayer player) {
+        final boolean success = writeToFile(file, format, schematic);
         final String message = success ? Names.Command.Save.Message.SAVE_SUCCESSFUL : Names.Command.Save.Message.SAVE_FAILED;
         player.sendMessage(new TextComponentTranslation(message, file.getName()));
     }
 
+    /**
+     * Gets a schematic format name translation key for the given format ID.
+     *
+     * Returns null if no such format exists, and logs a warning.
+     */
+    public static String getFormatName(final String id) {
+        if (!FORMATS.containsKey(id)) {
+            Reference.logger.warn("No format for schematic with id {}; returning null", new UnsupportedFormatException(id).fillInStackTrace(), id);
```

Comment:
Yea, looks like they are backwards.  I'll swap them.  (Though, I don't know if returning null is the right behavior here - would making `UnsupportedFormatException` unchecked and then throwing it be better?)